python/lsst/afw/cameraGeom/basicCameraFactory.py

```diff
@@ -0,0 +1,225 @@
+    amplifier information.  Cameras produced by this method should only be used for
+    mapping objects' positions on the sky to positions on the focal plane.
+
+    To create a camera, one should:
+
+    cameraFactory = BasicCameraFactory('path/to/focal/plane/layout.txt')
+    camera = cameraFactory.makeCamera()
+
+    The input text file should consist of one row for each detector.  Each row should contain
+
+    Name of the detector
+    The x coordinate of the detector's center in microns (relative to the center of the focal plane)
+    The y coordinate of the detector's center in microns
+    The size of each pixel in microns
+    The number of x pixels
+    The number of y pixels
+    A string indicating what type of detector it is (SCIENCE, FOCUS, GUIDER, or WAVEFRONT)
+    Three euler angles (in degrees) governing the orientation of the detector
+
+    For example:
+
+    #
+    # rows are the chips in the focal plane
+    # columns are the name, x position (microns), y position (microns),
+    # pixel size (microns), number of x pixels, number of y pixels,
+    # group of sensors, 3 euler rotations (degrees)
+    #
+    Det00 0.0 0.0 2.0 400 400 science 20.0 0.0 0.0
+    Det01 0.0 9000.0 2.0 400 400 science 10.0 0.0 0.0
+    Det02 -9000.0 0.0 2.0 400 400 wave 30.0 0.0 0.0
+    """
+
+    def __init__(self, detectorLayoutFile,
+                 detectorIdFromAbbrevName=None,
+                 detTypeMap=None,
+                 radialTransform=[0.0, 1.0, 0.0, 0.0],
+                 cameraName='LSST'
+                 ):
+
+        """
+        @param [in] detectorLayoutFile is the absolute path to the file
+        listing the layout of all of the chips in the camera's focal plane
+        (see this class' docstring for detailed information on the contents
+        of that file).
+
+        @param [in] detectorIdFromAbbrevName is a (required) method mapping the
+        names of detectors as stored in the detectorLayoutFile to unique integers
+        identifying each detector.
+
+        @param[in] detTypeMap is a dict mapping the group name of the sensors in
+        detectorLayoutFile to the integers:
+
+        0 = SCIENCE
+        1 = FOCUS
+        2 = GUIDER
+        3 = WAVEFRONT
+
+        if left as None, this will just try to cast the contents of detectorLayoutFile
+        as an int
+
+        @param [in] radialTransform is a list of coefficents that transform positions
+        from pupil coordinates (x, y, radians) to focal plane coordinates (x, y, mm)
+        according to the convention
+
+        (x, y)_focal = (x, y)_pupil * sum_i radialTransform[i] * r^i/r
+
+        where r is the magnitude of the point in pupil coordinates.  Note that the [1]
+        element of this list should be 1/plateScale where plateScale is in radians per mm.
+
+        @param [in] cameraName is a string referring to the name of the camera
+        (default 'LSST')
```

Comment:
Why default this to LSST?  You could make this a positional argument and make people name their cameras.